src/source/Signaling/Signaling.c

```diff
@@ -1173,3 +1208,39 @@ UINT64 signalingGetCurrentTime(UINT64 customData)
     UNUSED_PARAM(customData);
     return GETTIME();
 }
+
+STATUS signalingGetMetrics(PSignalingClient pSignalingClient, PSignalingClientMetrics pSignalingClientMetrics)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT64 curTime = GETTIME();
+
+    CHK(pSignalingClient != NULL && pSignalingClientMetrics != NULL, STATUS_NULL_ARG);
+    CHK(pSignalingClientMetrics->version <= SIGNALING_CLIENT_METRICS_CURRENT_VERSION, STATUS_SIGNALING_INVALID_METRICS_VERSION);
+
+    // Interlock the threading due to data race possibility
+    MUTEX_LOCK(pSignalingClient->diagnosticsLock);
+
+    // Fill in the data structures according to the version of the requested structure - currently only v0
+    pSignalingClientMetrics->signalingClientStats.signalingClientUptime = curTime - pSignalingClient->diagnostics.createTime;
+    pSignalingClientMetrics->signalingClientStats.numberOfMessagesSent = (UINT32) pSignalingClient->diagnostics.numberOfMessagesSent;
+    pSignalingClientMetrics->signalingClientStats.numberOfMessagesReceived = (UINT32) pSignalingClient->diagnostics.numberOfMessagesReceived;
+    pSignalingClientMetrics->signalingClientStats.iceRefreshCount = (UINT32) pSignalingClient->diagnostics.iceRefreshCount;
+    pSignalingClientMetrics->signalingClientStats.numberOfDynamicErrors = (UINT32) pSignalingClient->diagnostics.numberOfDynamicErrors;
+    pSignalingClientMetrics->signalingClientStats.numberOfReconnects = (UINT32) pSignalingClient->diagnostics.numberOfReconnects;
+    pSignalingClientMetrics->signalingClientStats.cpApiCallLatency = pSignalingClient->diagnostics.cpApiLatency;
+    pSignalingClientMetrics->signalingClientStats.dpApiCallLatency = pSignalingClient->diagnostics.dpApiLatency;
+
+    if (ATOMIC_LOAD_BOOL(&pSignalingClient->connected)) {
+        pSignalingClientMetrics->signalingClientStats.connectionDuration = curTime - pSignalingClient->diagnostics.connectTime;
+    } else {
+        pSignalingClientMetrics->signalingClientStats.connectionDuration = INVALID_TIMESTAMP_VALUE;
```

Comment:
I think it's better if it's set to 0 instead. Let say we want to dump these metrics to Cloudwatch, dumping it as 0 will give more meaningful data, we can infer that the device is not connected to our signaling. However, if we use `INVALID_TIMESTAMP_VALUE`, it is ambiguous, it can be that there's a device that has run for `0xFFFFFFFFFFFFFFFF` (it's definitely unlikely) or the metrics get aggregated.

Reply:
You are absolutely correct. Will do that